### ISSUE TITLE
AuthAD plugin: group/domain membership in multi-domain when SSO enabled.

### ISSUE THREAD

**Issue by Lord-Dimwit-Flathead-the-Excessive** [splitbrain/dokuwiki#1736]
Synopsis: When SSO = 1, users that are **not** members of the primary domain can connect but Dokuwiki treats them as if they logged in from the primary domain.  When SSO = 0 and users log in using the domain selector, group membership seems to work for both domains.

Example (the names have been changed to protect the infrastructure):  

jack@primary.domain.org belongs to no groups other than "Domain Users"
jack@secondary.domain.org is member of group wikiAdmins in secondary.domain.org
both have unique AD passwords.
$conf['superuser'] = '@WikiAdmins'
$conf["plugin']['authad']['sso'] = 1;

secondary\jack browses to Dokuwiki and is logged on without admin privs and as "(jack)"; not as "(jack@secondary)" as is the case when sso = 0 and the domain selector log-in page is used.

Example 2:

jack@secondary.domain.org belongs to no groups other than "Domain Users"
jack@primary.domain.org is member of group wikiAdmins in primary.domain.org
both have unique AD passwords.
$conf['superuser'] = '@WikiAdmins'
$conf["plugin']['authad']['sso'] = 1;

secondary\jack browses to Dokuwiki and is logged **with** admin privs and as "(jack)"; not as "(jack@secondary)" as should be.

Possibly Related: Admin + User Manager only enumerates users from the primary domain.

**Comment by splitbrain** [comment]
What type of SSO-mechanism are you using? Are you sure your domain is properly passed?

**Comment by Lord-Dimwit-Flathead-the-Excessive** [comment]
Wow! You really get back quick!  

SSO: Just browsers capable of Windows Integrated Authentication (testing using both Chrome and IE) and the Kerberos installed per your procedure https://www.dokuwiki.org/plugin:authad.  And SSO works fine as long as you are logging in from the primary domain.

I'm not exactly certain what you're asking in your second question.    Do you have a command line test you want me to run?  Or I can plop a msg() debug line anywhere you like in auth.php and connect;  I've been doing a bit of that trying to figure this one out...

**Comment by Lord-Dimwit-Flathead-the-Excessive** [comment]
Yup.  But for both domains it passes in the format: username@DOMAIN.COM and cleanUser returns just username (lowercased).  The domains are full uppercase.  Are they actually the Kerberos Realms?  Also, you should be aware that one of the actual domain names we are dealing with has a hyphen in it and the other ends with a non-standard namespace.  For security reasons, I don't want to publish the actual domain and usernames here but I'll be happy to share the actual outputs on a "back channel" with you.
